src/function_classes/mlp.py

```diff
@@ -21,7 +21,7 @@ def _init_param_dist(self) -> D.Distribution:
         return param_dist
 
     def evaluate(self, x_batch: torch.Tensor, *params: torch.Tensor) -> torch.Tensor:
-        raw_tensor, _ = params
+        raw_tensor = torch.stack(params)
```

Comment:
here, `params` is of length 32, so the weights are precisely params, so calling `torch.stack` correctly gets the weights.